js/web/lib/wasm/wasm-training-core-impl.ts

```diff
@@ -313,6 +317,131 @@ export const runTrainStep = async(
   }
 };
 
+export const getParametersSize = (trainingSessionId: number, trainableOnly: boolean): number => {
+  const wasm = getInstance();
+  const stack = wasm.stackSave();
+
+  try {
+    const sizeOffset = wasm.stackAlloc(4);
+    if (wasm._OrtTrainingGetParametersSize) {
+      const errorCode = wasm._OrtTrainingGetParametersSize(trainingSessionId, sizeOffset, trainableOnly);
+      ifErrCodeCheckLastError(errorCode, 'Can\'t get parameters size');
+
+      return wasm.HEAP32[sizeOffset / 4];
+    } else {
+      throw new Error(NO_TRAIN_FUNCS_MSG);
+    }
+  } finally {
+    wasm.stackRestore(stack);
+  }
+};
+
+export const getContiguousParameters =
```

Comment:
This can be part of a separate PR. I see there are lot of mtds which allocate data, it will be good to add some utils which can be called form everywhere, for example prepareInputOutputTensor also does a lot of this, not sure if you can simply call this mtd, but can do some refactoring to reduce duplication.